src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3982,14 +3989,184 @@ bool   os::win32::_is_windows_server         = false;
+    if (last_error == ERROR_INSUFFICIENT_BUFFER) {
+      DWORD group_count = job_object_information_length / sizeof(GROUP_AFFINITY);
+
+      job_object_information = os::malloc(job_object_information_length, mtInternal);
+      if (job_object_information != nullptr) {
+          if (QueryInformationJobObject(nullptr, JobObjectGroupInformationEx, job_object_information, job_object_information_length, &job_object_information_length)) {
+            DWORD groups_found = job_object_information_length / sizeof(GROUP_AFFINITY);
+            if (groups_found != group_count) {
+              warning("Unexpected processor group count->%ld. Expected %ld processor groups.", groups_found, group_count);
+              assert(false, "Unexpected group count");
+            }
+
+            GROUP_AFFINITY* group_affinity_data = ((GROUP_AFFINITY*)job_object_information);
+            for (DWORD i = 0; i < groups_found; i++) {
+              processors += population_count(group_affinity_data[i].Mask);
+            }
+
+            if (processors == 0) {
+              warning("Could not determine processor count from the job object.");
+              assert(false, "Must find at least 1 logical processor");
+            }
+          } else {
+            warning("QueryInformationJobObject() failed: GetLastError->%ld.", GetLastError());
+          }
+
+          os::free(job_object_information);
+      } else {
+          warning("os::malloc() failed to allocate %ld bytes for QueryInformationJobObject", job_object_information_length);
+      }
+    }
+  }
+
+  return processors;
+}
+
+DWORD os::win32::system_logical_processor_count() {
+  DWORD logical_processors = 0;
+  LOGICAL_PROCESSOR_RELATIONSHIP relationship_type = RelationGroup;
+  PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX system_logical_processor_info = nullptr;
+  DWORD returned_length = 0;
+
+  // https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformationex
+  if (!GetLogicalProcessorInformationEx(relationship_type, nullptr, &returned_length)) {
+    DWORD last_error = GetLastError();
+
+    if (last_error == ERROR_INSUFFICIENT_BUFFER) {
+      system_logical_processor_info = (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)os::malloc(returned_length, mtInternal);
+
+      if (nullptr == system_logical_processor_info) {
+        warning("os::malloc() failed to allocate %ld bytes for GetLogicalProcessorInformationEx buffer", returned_length);
+      } else if (!GetLogicalProcessorInformationEx(relationship_type, system_logical_processor_info, &returned_length)) {
+        warning("GetLogicalProcessorInformationEx() failed: GetLastError->%ld.", GetLastError());
+      } else {
+        DWORD processor_groups = system_logical_processor_info->Group.ActiveGroupCount;
+
+        PROCESSOR_GROUP_INFO* group_info = (PROCESSOR_GROUP_INFO*)system_logical_processor_info->Group.GroupInfo;
+        for (DWORD i = 0; i < processor_groups; i++) {
```

Comment:
same here you dont need the dereference if you increment the pointer after 'i++, group_info++' and then 'group_info->ActiveProcessorCount'

Reply:
> same here you dont need the dereference if you increment the pointer after 'i++, group_info++' and then 'group_info->ActiveProcessorCount'

The same pointer address is dereferenced using either style. I will leave it as is unless I get additional feedback upstream about this.

Reply:
It's not a style thing! The current code does more math (an additional multiply), you have spent more time checking that all the compilers will make the change for you than it would take to change the code.

nit: I strongly suggest to other C++ developers to follow the method where you increment the pointer